src/pages-and-resources/teams/TeamSetEditor.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import * as Yup from 'yup';
+
+import {
+  Button, Collapsible, TransitionReplace, Form,
+} from '@edx/paragon';
+import { useFormik } from 'formik';
+import messages from './messages';
+
+function TeamSetEditor({
+  intl, teamSet, onDelete,
+}) {
+  const [isDeleting, setDeleting] = useState(false);
+  const initiateDeletion = () => setDeleting(true);
+  const cancelDeletion = () => setDeleting(false);
+
+  const {
+    handleChange,
+    handleBlur,
+    values,
+    // errors,
+  } = useFormik({
+    initialValues: teamSet,
+    validationSchema: Yup.object().shape({
+      name: Yup.string().required(),
+      description: Yup.string(),
+      type: Yup.mixed().oneOf(['open', 'public_managed', 'private_managed']).required(),
+      max_team_size: Yup.number().required(),
+    }),
+    // onSubmit,
+  });
+
+  return (
+    <Collapsible
+      title={(
+        <div className="d-flex flex-column small">
+          <span>Open</span>
+          <span className="h4">{teamSet.name}</span>
+          <span>{teamSet.description}</span>
+        </div>
+    )}
+    >
+      <TransitionReplace>
+        {isDeleting
+          ? (
+            <div className="d-flex flex-column m-4" key="isDeleting">
+              <h3>{intl.formatMessage(messages.team_set_delete_heading)}</h3>
+              <p>{intl.formatMessage(messages.team_set_delete_body)}</p>
+              <div className="d-flex flex-row justify-content-end">
+                <Button variant="muted" size="sm" onClick={() => onDelete(teamSet)}>
+                  {intl.formatMessage(messages.delete)}
+                </Button>
+                <Button variant="muted" size="sm" onClick={cancelDeletion}>
+                  {intl.formatMessage(messages.cancel)}
+                </Button>
+              </div>
+            </div>
+          )
```

Comment:
Suggestion: this should be in generics to be used in other components as well. this is already in use at other places in the application.

Reply:
I'm not sure what exactly you're referring to here?